src/Nethermind/Nethermind.Consensus/Processing/BranchProcessor.cs

```diff
@@ -172,7 +166,9 @@ public Block[] Process(BlockHeader? baseBlock, IReadOnlyList<Block> suggestedBlo
                 {
                     if (_logger.IsInfo) _logger.Info($"Commit part of a long blocks branch {i}/{blocksCount}");
                     previousBranchStateRoot = suggestedBlock.Header;
-                    InitBranch(suggestedBlock.Header, false);
+
+                    worldStateCloser.Dispose();
```

Comment:
not relevant right now - but, this is good, we can add another check here, for something like IsTransitionBlock